test/cri-containerd/runpodsandbox_test.go

```diff
@@ -1137,6 +1138,93 @@ func Test_RunPodSandbox_Mount_SandboxDir_LCOW(t *testing.T) {
 	//TODO: Parse the output of the exec command to make sure the uvm mount was successful
 }
 
+func Test_RunPodSandbox_CPUGroup_LCOW(t *testing.T) {
+	requireFeatures(t, featureLCOW)
+
+	pullRequiredLcowImages(t, []string{imageLcowK8sPause})
+
+	ctx := context.Background()
+
+	presentID := "FA22A12C-36B3-486D-A3E9-BC526C2B450B"
+	// we believe it is a resonable assumption that our test machines have at least two cores
+	// that can we used to create a cpugroup
+	logicalProcessors := []uint32{0, 1}
+	if err := uvm.CreateNewCPUGroupWithID(ctx, presentID, logicalProcessors); err != nil {
+		t.Fatalf("failed to create test cpugroup with: %v", err)
+	}
+
+	defer func() {
+		err := uvm.DeleteCPUGroup(ctx, presentID)
+		if err != nil && err != uvm.HV_STATUS_INVALID_CPU_GROUP_STATE {
+			t.Fatalf("failed to clean up test cpugroup with: %v", err)
+		}
+	}()
+
+	annotations := []map[string]string{
+		{
+			"io.microsoft.virtualmachine.cpugroup.id": presentID,
+		},
+	}
+
+	for _, a := range annotations {
+		request := &runtime.RunPodSandboxRequest{
+			Config: &runtime.PodSandboxConfig{
+				Metadata: &runtime.PodSandboxMetadata{
+					Name:      t.Name(),
+					Uid:       "0",
+					Namespace: testNamespace,
+				},
+				Annotations: a,
+			},
+			RuntimeHandler: lcowRuntimeHandler,
+		}
+		runPodSandboxTest(t, request)
+	}
+}
+
+func Test_RunPodSandbox_CPUGroup_WCOW_Hypervisor(t *testing.T) {
+	requireFeatures(t, featureWCOWHypervisor)
+
+	pullRequiredImages(t, []string{imageWindowsNanoserver})
+	ctx := context.Background()
+
+	presentID := "FA22A12C-36B3-486D-A3E9-BC526C2B450B"
+	// we believe it is a resonable assumption that our test machines have at least two cores
```

Comment:
Could we check and skip this test if this assertion does not hold?

Reply:
Addressed below, but updated.